Mathlib/SetTheory/Nimber/Field.lean

```diff
@@ -218,4 +217,127 @@ instance : CommRing Nimber where
 instance : IsDomain Nimber where
 instance : CancelMonoidWithZero Nimber where
 
+/-! ### Nimber division -/
+
+mutual
+
+/-- The nimber inverse `a⁻¹` is recursively defined as the smallest nimber not in the set `s`, which
+itself is recursively defined as the smallest set with `0 ∈ s` and `(1 + (a + a') * b) / a' ∈ s`
+for `0 < a' < a` and `b ∈ s`.
+
+This preliminary definition "accidentally" satisfies `invAux 0 = 1`, which the real inverse
+corrects. -/
+def invAux (a : Nimber) : Nimber :=
+  sInf (invSet a)ᶜ
+termination_by (a, 1)
+
+/-- The set in the definition of `invAux a`. -/
+def invSet (a : Nimber) : Set Nimber :=
+  ⋂₀ {s | 0 ∈ s ∧ ∀ a' < a, a' ≠ 0 → ∀ b ∈ s, invAux a' * (1 + (a + a') * b) ∈ s}
+termination_by (a, 0)
```

Comment:
```suggestion
/-- The set in the definition of `invAux a`. -/
def invSet (a : Nimber) : Set Nimber :=
  ⋂₀ {s | 0 ∈ s ∧ ∀ a' < a, a' ≠ 0 → ∀ b ∈ s, sInf (invSet a')ᶜ * (1 + (a + a') * b) ∈ s}
termination_by (a, 0)
```
What breaks if I try to define it like this? Doing it this way seems cleaner imo, so you can avoid the mutual recursion. But I can believe that something breaks down the line - if so let's add a comment explaining that.

Reply:
This is very similar to what I initially had, it was @eric-wieser's suggestion to use mutual induction.

I do prefer the mutual induction approach since it gives nicer results when doing `rw [invSet]` or `rw [invAux]`.

Reply:
I think the key distinction is that this version means you only need to state the set once, rather than twice in your original. But yeah the mutual induction is nice enough